src/PowerShellEditorServices/Utility/ArgumentUtils.cs

```diff
@@ -0,0 +1,37 @@
+using System.Text;
+using System.Management.Automation.Language;
+
+namespace Microsoft.PowerShell.EditorServices.Utility
+{
+    internal static class ArgumentEscaping
+    {
+        /// <summary>
+        /// Escape a powershell argument while still making it evaluatable in AddScript.
+        ///
+        /// NOTE: This does not "santize" parameters, e.g. a pipe in one argument might affect another argument.
+        /// This is intentional to give flexibility to specifying arguments.
+        /// It also does not try to fix invalid PowerShell syntax, e.g. a single quote in a string literal.
+        /// </summary>
+        public static string Escape(string Arg)
+        {
+            // if argument is a scriptblock return as-is
+            if (Arg.StartsWith("{") && Arg.EndsWith("}"))
```

Comment:
@JustinGrote Should we strip whitespace first? In case it's ` { ... } ` or some other variation with (pointless) spaces before/after.

Reply:
> @JustinGrote Should we strip whitespace first? In case it's `{ ... }` or some other variation with (pointless) spaces before/after.

I think this is another case of "overly helpful", if you specified your debug config and left a space at the beginning or the end before your scriptblock, I have to think you may have had an intention there.